controllers/nodepools.go

```diff
@@ -0,0 +1,139 @@
+package controllers
```

Comment:
A nitpick - Can you add license headers where missing?

Reply:
Added now, thanks very much @sutaakar :)